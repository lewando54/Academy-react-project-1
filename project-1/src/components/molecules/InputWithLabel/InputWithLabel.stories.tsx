/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import InputWithLabel from './InputWithLabel'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: InputWithLabel,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    id: { control: 'text' },
    name: { control: 'text' },
    type: { control: 'select', options: ['text', 'mail', 'password', 'date'] },
    children: { control: 'text' }
  }
} as ComponentMeta<typeof InputWithLabel>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof InputWithLabel> = (args) => <InputWithLabel {...args} />

export const Primary = Template.bind({})

Primary.args = {
  children: 'Example label',
  id: 'input-id',
  name: 'input-name',
  type: 'text'
}
// More on args: https://storybook.js.org/docs/react/writing-stories/args
