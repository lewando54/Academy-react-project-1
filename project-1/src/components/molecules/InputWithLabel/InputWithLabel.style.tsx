import styled from 'styled-components'

export default styled.div`
    display: flex;
    flex-direction: column;
    gap: 0.25rem;
    width: 100%;
`
