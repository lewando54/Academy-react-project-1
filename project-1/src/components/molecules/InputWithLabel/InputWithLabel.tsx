import React, { forwardRef } from 'react'
import Input, { InputProps } from '../../atoms/Input/Input'
import Label, { LabelProps } from '../../atoms/Label/Label'
import StyledInputWithLabel from './InputWithLabel.style'

type InputWithLabelProps = InputProps & LabelProps

function InputWithLabel (props: InputWithLabelProps, ref: any): React.ReactElement {
  return (
        <StyledInputWithLabel>
            <Label htmlFor={props.id}>{props.children}</Label>
            <Input type={props.type} id={props.id} name={props.name} ref={ref} required={props.required}/>
        </StyledInputWithLabel>
  )
}

// imperative tutaj może rozszerzyć

export default forwardRef(InputWithLabel)
