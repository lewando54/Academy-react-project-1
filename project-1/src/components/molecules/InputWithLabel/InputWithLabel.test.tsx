import React from 'react'
import '@testing-library/jest-dom'
import * as renderer from 'react-test-renderer'
import InputWithLabel from './InputWithLabel'

describe('InputWithLabel', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<InputWithLabel htmlFor='checkid' id='checkid' name='test' type='text' testId='test'>Testing</InputWithLabel>)
    expect(tree).toMatchSnapshot()
  })
})
