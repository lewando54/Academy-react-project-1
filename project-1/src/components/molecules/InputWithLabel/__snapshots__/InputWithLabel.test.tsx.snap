// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel should render correctly 1`] = `
<div
  className="sc-gswNZR bYjhy"
>
  <label
    data-testid="test"
    htmlFor="checkid"
  >
    Testing
  </label>
  <input
    className="sc-bcXHqe gBKJnZ"
    data-testid="test"
    id="checkid"
    name="test"
    type="text"
  />
</div>
`;
